### ISSUE TITLE
Natron cannot be used to edit videos

### ISSUE THREAD

**Issue by ocket8888** [NatronGitHub/Natron#555]
### Problem
**Expected behavior:** Natron to not re-order frames unless I tell it to

**Actual behavior:** Natron cannot be used to output any kind of comprehensible video, as itre-orders frames in the video if they are previewed

### Steps to Reproduce

1. I have a simple rendering pipeline that does literally nothing to the video before writing it:

    ![image](https://user-images.githubusercontent.com/6013378/100660360-27c9e180-330f-11eb-91dd-5709731e4edd.png)

1. Then, I click "play" and watch it stagger through the first 230-ish frames of the [source video](https://streamable.com/h96c3t).

1. Finally, I write the [output video](https://streamable.com/ohunkv) and observe that it stutters and appears to have randomized the order of the previewed frames.

This is a fresh install, first time I've ever launched Natron.

### Versions

* **Natron version/commit** 2.3.15 Release
* **OS version:** Windows 10

**Comment by rodlie** [comment]
Your source video codec is most likely not supported properly, also note that Natron is not an video editor.

**Comment by ocket8888** [comment]
I'd be surprised to hear it doesn't support `.mkv`, but I don't see anything about supported codecs in the docs, so I'll take your word for it.

> _"... note that Natron is not an video editor."_

So I want to add some laser beams to this video - Natron is not capable of doing this?

**Comment by rodlie** [comment]
MKV is a container, not a codec. I'm able to read various MKV's without any major issues, so it depends on the codec used in the container (MKV).

> So I want to add some laser beams to this video - Natron is not capable of doing this?

Yes, but in your case probably not the best solution. Natron is a compositor, designed for image sequences.

**Comment by ocket8888** [comment]
> _"MKV is a container, not a codec."_

How do I find out what codec it is? Could something like VLC tell me or is that information unavailable in the .mkv itself?

> _"... in your case probably not the best solution. Natron is a compositor, designed for image sequences."_

I've done laser beams before in After Effects, so I just chose the first alternative on [alternativeto.net's free-and-open-source alternatives list for After Effects](https://alternativeto.net/software/adobe-after-effects/?license=opensource), assuming that it'd be comparable in function. I don't suppose you have better suggestion? I really don't want to have to go back to closed-source if I don't have to.

**Comment by devernay** [comment]
So when you press play inside Natron, are the frames shuffled, or are they in the right order?
If they are in the right order, then the output codec is the problem.
Documentation:
- https://natron.readthedocs.io/en/rb-2.3/guide/getstarted-troubleshooting.html#common-workarounds
- https://natron.readthedocs.io/en/rb-2.3/guide/tutorials-imagesequence.html
- https://natron.readthedocs.io/en/rb-2.3/guide/tutorials-imagesequence.html#how-to-convert-image-sequences-to-video-files

**Comment by ocket8888** [comment]
Inside Natron they appear to be out-of-order as well. Initially, though, I thought it was just having trouble rendering the video in real-time, so I only took notice of it in the output.

**Comment by Shrinks99** [comment]
Just a clarification, Natron is not a [video editing program](https://en.wikipedia.org/wiki/Non-linear_editing), it is a [compositing program](https://en.wikipedia.org/wiki/Compositing).  You've got the right program for your use case (adding lasers) but just the wrong terms to describe it.

You will have much better luck converting your MKV to an image sequence and then using that as the source in Natron.  Video files do not perform reliably or quickly in this software.

**Comment by ocket8888** [comment]
won't that destroy the audio source? I can obviously just rip the audio from the original mkv and impose it on the output video, but that seems like a bunch of extra steps.

It doesn't really matter for this use-case, but could be a pain in the future

**Comment by Shrinks99** [comment]
Yep.  For any actual production you would export the shot from your editing program as frames over to Natron and then re-import it back into the timeline.  Nobody really cares about audio in comp because it's dealt with by editorial later on in the post production process.

**Comment by ocket8888** [comment]
Well then maybe I'll give that a shot. There's probably a way for OBS to output separate audio and video tracks anyway, to let me skip a step - or at least a plugin that can do it.

So it seems the only real issue here is that I'm using an unsupported codec. Since it's the default codec used by OBS there might be interest in supporting it - if you tell me how to find out what codec that is I can change this issue to a request for its support as input.

**Comment by securingmom** [comment]
@ocket8888 

> find out what [encoder]

**[MediaInfo](https://mediaarea.net/en/MediaInfo)**.  It is a cross platform tool.  Additionally this site might have more of the answers you seek

https://videohelp.com/search?q=codec+info&siteurl=&Search=Search
https://videohelp.com/software/MediaInfo

**Comment by ocket8888** [comment]
@securingmom Thanks. It took me a while, but this issue is now a request for AVC support

**Comment by ThatOneCoderYouLove** [comment]
Tbh I just switched to mp4v codec and it worked, not a direct solution to avc not working but an alternative

**Comment by ThatOneCoderYouLove** [comment]
> Tbh I just switched to mp4v codec and it worked, not a direct solution to avc not working but an alternative

Broke again, had to switch to AV1
